appeals/web/src/server/appeals/appeal-details/interested-party-comments/interested-party-comments.mapper.js

```diff
@@ -75,35 +75,132 @@ function generateTableRows(items, isReview = false) {
+	};
+
+	return pageContent;
+}
+
+/**
+ * @param {Appeal} appealDetails
+ * @param {Representation} comment
+ * @returns {PageContent}
+ */
+export function reviewInterestedPartyCommentPage(appealDetails, comment) {
+	const shortReference = appealShortReference(appealDetails.appealReference);
+	const commentSummaryList = generateCommentSummaryList(comment);
+	const withdrawLink = generateWithdrawLink();
+
+	/** @type {PageComponent} */
+	const siteVisitRequestCheckbox = {
+		type: 'checkboxes',
+		parameters: {
+			name: 'site-visit-request',
+			items: [
+				{
+					text: 'The comment includes a site visit request',
+					value: 'site-visit',
+					checked: false
+				}
+			]
+		}
+	};
 
-	const attachmentsList = (() => {
-		if (comment.attachments.length === 0) {
-			return null;
+	/** @type {PageComponent} */
+	const commentValidityRadioButtons = {
+		type: 'radios',
+		parameters: {
+			name: 'status',
+			fieldset: {
+				legend: {
+					text: 'Do you accept the comment?',
+					isPageHeading: false,
+					classes: 'govuk-fieldset__legend--m'
+				}
+			},
+			items: [
+				{
+					value: 'valid',
+					text: 'Comment valid',
+					checked: comment?.status === 'valid'
+				},
+				{
+					value: 'invalid',
+					text: 'Comment invalid',
+					checked: comment?.status === 'invalid'
+				}
+			]
 		}
+	};
+
+	const pageContent = {
+		title: 'Review comment',
+		backLinkUrl: `/appeals-service/appeal-details/${appealDetails.appealId}/interested-party-comments`,
+		preHeading: `Appeal ${shortReference}`,
+		heading: 'Review comment',
+		headingClasses: 'govuk-heading-l',
+		submitButtonText: 'Confirm',
+		pageComponents: [
+			commentSummaryList,
+			siteVisitRequestCheckbox,
+			commentValidityRadioButtons,
+			withdrawLink
+		]
+	};
 
-		const items = comment.attachments.map(
-			(a) => `
-      <li>
-        <a href="#">${a.documentVersion.document.name}</a>
-      </li>
-    `
-		);
+	return pageContent;
+}
 
-		return `<ul class="govuk-list">${items.join('')}</ul>`;
-	})();
+/**
+ * Generates the withdraw link component.
+ * @returns {PageComponent} The withdraw link component.
+ */
+function generateWithdrawLink() {
+	return {
+		type: 'html',
+		wrapperHtml: {
+			opening: '<div class="govuk-!-margin-bottom-3">',
+			closing: '</div>'
+		},
+		parameters: {
+			html: '<a class="govuk-link" href="#">Withdraw comment</a>'
+		}
+	};
+}
 
+/**
+ * Generates the comment summary list used in both view and review pages.
+ * @param {Representation} comment - The comment object.
+ * @returns {PageComponent} The generated comment summary list component.
+ */
+function generateCommentSummaryList(comment) {
 	const hasAddress =
 		comment.represented.address &&
 		comment.represented.address.addressLine1 &&
 		comment.represented.address.postCode;
 
 	const commentIsDocument = !comment.originalRepresentation && comment.attachments?.length > 0;
 
-	/** @type {PageComponent} */
-	const commentSummaryList = {
+	const attachmentsList =
```

Comment:
`buildHtmUnorderedList` in `appeals/web/src/server/lib/nunjucks-template-builders/tag-builders.js` might be useful here